pkg/kube/watcher.go

```diff
@@ -18,12 +19,33 @@ var (
 		Name: "events_sent",
 		Help: "The total number of events sent",
 	})
+	eventsCount = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "events_count",
+		Help: "The current number of event sent(server side)",
+	}, []string{"namespace", "name", "involved_object_namespace", "involved_object_name", "involved_object_kind", "reason", "type", "source"})
+	eventsTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "events_total",
+		Help: "The total number of events received(watcher)",
+	}, []string{"namespace", "name", "involved_object_namespace", "involved_object_name", "involved_object_kind", "reason", "type", "source"})
 	watchErrors = promauto.NewCounter(prometheus.CounterOpts{
 		Name: "watch_errors",
 		Help: "The total number of errors received from the informer",
 	})
 )
 
+func getMetricValues(event *corev1.Event) []string {
+	return []string{
+		event.ObjectMeta.Namespace,
+		event.ObjectMeta.Name,
+		event.InvolvedObject.Namespace,
+		event.InvolvedObject.Name,
+		event.InvolvedObject.Kind,
+		event.Reason,
+		event.Type,
+		fmt.Sprintf("%s/%s", event.Source.Host, event.Source.Component),
```

Comment:
is it necessary to concatenate these two (host and component)?  Doing so adds overhead to every event.  If you pass them through like the above strings, the data will still be available in the labels, without any per-event overhead.  On big clusters there can be well over 1k events per second, and avoiding any extra work in that process will save a lot of resources over time.

Reply:
maybe make the `[]string` only once and pass it twice.